### PR TITLE
[WIP] Addressed build issues

### PR DESCRIPTION
**Pull request by ajbozarth**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,12 +41,15 @@ clean: ## Make a clean source tree
...
 	rm -rf $$(find . -name .pytest_cache)
-
-# Prepares Elyra for build/packaging/installation
-yarn-install:
-	rm -f yarn.lock package-lock.json
 	yarn cache clean
-	yarn
+	pip uninstall -y jupyterlab-git
+	- jupyter labextension uninstall --no-build @jupyterlab/toc
+	$(call UNLINK_LAB_EXTENSION,application)
```
Comment (lresende) [inline Makefile:47]: Could we have an `unlink` target and make `clean` depend on it. Do we need to have `install` also depend on it in case of the user only issuing a `make install` ?

File: Makefile [hunk 2]
```diff
@@ -78,17 +81,25 @@ bdist: npm-packages
...
 	$(call LINK_LAB_EXTENSION,application)
-	$(call LINK_LAB_EXTENSION,theme)
-	$(call LINK_LAB_EXTENSION,code-snippet)
-	$(call LINK_LAB_EXTENSION,pipeline-editor)
-	$(call LINK_LAB_EXTENSION,python-runner)
+	jupyter lab build
+	jupyter serverextension list
+	jupyter labextension list
+
+install-backend: ## Build and install backend
```
Comment (lresende) [inline Makefile:89]: can we call this `install-server` to be consistent with `test-server`? The other way around is also ok.

File: Makefile [hunk 3]
```diff
@@ -78,17 +81,25 @@ bdist: npm-packages
 
 install: bdist lint ## Build distribution and install
 	pip install --upgrade dist/elyra-*-py3-none-any.whl
```
Comment (lresende) [inline Makefile:83]: The original idea here would be to enable handing a wheel to someone and he would be able to install/update elyra on his system, but after our conversation, it does not seem that this is possible unless the npm packages are published. We should have a conversation and work out what should we do and maybe get rid of these.

File: Makefile (hunk 1 above)
Comment (ajbozarth) [reply Makefile:47]: I did have a separate target but I didn't like the order it was executing so I switched back to all of it in one target. I have no string opinions on it though and can separate them again.

As for make install, I can test some more edge cases but I don't believe make install needs to depend on clean anymore, I believe that was a build issue in 1.x that was fixed, but again I'd need to do some more edge case testing to know for sure. @vabarbosa if you're willing to help test the edge cases since you've also been working on some of this.

File: Makefile (hunk 2 above)
Comment (ajbozarth) [reply Makefile:89]: Sure, I like that better as well. (This was an already existing target that I just reused so I didn't want to edit it)

File: Makefile (hunk 3 above)
Comment (ajbozarth) [reply Makefile:83]: Based on what we discuss in scrum, I have a couple ideas that would remove quite a bit from the make file and streamline this pr

File: Makefile (hunk 1 above)
Comment (vabarbosa) [reply Makefile:47]: count me in

**Comment by ajbozarth** [comment]
Closing in deference to #548
